source/ref_gl/r_renderer_api.h

```diff
@@ -0,0 +1,26 @@
+#ifndef R_RENDERER_API_H
+#define R_RENDERER_API_H
+
+#include "../gameshared/q_shared.h"
+#include "r_math.h"
+
+typedef struct image_s image_t;
+
+DECLARE_STUB_IMPL( void, R_BindFrameBufferObject, int object );
+DECLARE_STUB_IMPL( void, R_Set2DMode, bool enable ); 
+DECLARE_STUB_IMPL( void, R_SetGamma, float gamma );
+DECLARE_STUB_IMPL( void, R_Scissor, int x, int y, int w, int h );
+DECLARE_STUB_IMPL( void, R_GetScissor, int *x, int *y, int *w, int *h );
+DECLARE_STUB_IMPL( void, R_ResetScissor, void );
+DECLARE_STUB_IMPL( void, R_DrawStretchQuick, int x, int y, int w, int h, float s1, float t1, float s2, float t2, const vec4_t color, int program_type, image_t *image, int blendMask );
+DECLARE_STUB_IMPL( void, R_ClearRefInstStack, void );
+DECLARE_STUB_IMPL( bool, R_PushRefInst, void );
+DECLARE_STUB_IMPL( void, R_PopRefInst, void );
+DECLARE_STUB_IMPL( void, R_DeferDataSync, void );
+DECLARE_STUB_IMPL( void, R_DataSync, void );
+DECLARE_STUB_IMPL( void, R_Finish, void );
+DECLARE_STUB_IMPL( void, R_Flush, void );
```

Comment:
my plan is to stub all these out then implement the new backend off these API calls. I'm mainly interested in anything between this and the rendering backend. there is a clear split in the API that should make this a good place to implement. most of the details need to be copied but I don't think the duplications will be a problem. vulkan is different enough where trying to fit vulkan onto this API will pose a challenge.